aspnetcore/fundamentals/environments.md

```diff
@@ -1,601 +1,580 @@
+{
+    EnvironmentName = Environments.Staging
+}); 
+```
 
-To set the `ASPNETCORE_ENVIRONMENT` environment variable with `web.config`, see the *Set environment variables* section of <xref:host-and-deploy/iis/web-config#set-environment-variables>.
+:::moniker-end
 
-**Project file or publish profile**
+:::moniker range="< aspnetcore-6.0"
 
-**For Windows IIS deployments:** Include the `<EnvironmentName>` property in the [publish profile (.pubxml)](xref:host-and-deploy/visual-studio-publish-profiles) or project file. This approach sets the environment in *web.config* when the project is published:
+Call <xref:Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.UseEnvironment%2A> when building the host. For more information, see <xref:fundamentals/host/generic-host#environmentname>.
 
-```xml
-<PropertyGroup>
-  <EnvironmentName>Development</EnvironmentName>
-</PropertyGroup>
-```
+:::moniker-end
 
-**Per IIS Application Pool**
+## Load configuration by environment
 
-To set the `ASPNETCORE_ENVIRONMENT` environment variable for an app running in an isolated Application Pool (supported on IIS 10.0 or later), see the *AppCmd.exe command* section of the [Environment Variables &lt;environmentVariables&gt;](/iis/configuration/system.applicationHost/applicationPools/add/environmentVariables/#appcmdexe) topic. When the `ASPNETCORE_ENVIRONMENT` environment variable is set for an app pool, its value overrides a setting at the system level.
+To load configuration by environment, see <xref:fundamentals/configuration/index#json-configuration-provider>.
 
-When hosting an app in IIS and adding or changing the `ASPNETCORE_ENVIRONMENT` environment variable, use any one of the following approaches to have the new value picked up by apps:
+## Environment-based `Startup` class and methods
 
-* Execute `net stop was /y` followed by `net start w3svc` from a command prompt.
-* Restart the server.
+Since the release of .NET 6, the ASP.NET Core project templates place startup code for service configuration and request pipeline processing in the `Program` file. Use of a `Startup` class (`Startup.cs`) with [`Configure`](xref:Microsoft.AspNetCore.Hosting.StartupBase.Configure%2A) and [`ConfigureServices`](xref:Microsoft.AspNetCore.Hosting.StartupBase.ConfigureServices%2A) methods was required before the release of .NET 6 but remains supported in ASP.NET Core. You can use the guidance in this section to control code execution by environment using a `Startup` class in any release of ASP.NET Core when a large number of code changes are required for each environment in use by an app.
 
-#### macOS
+Except for the [`Startup` class conventions](#startup-class-conventions) approach, the `Program` class uses the following general approach to identify the `Startup` class, where the `Startup` class is the type for the <xref:Microsoft.AspNetCore.Hosting.WebHostBuilderExtensions.UseStartup%2A?displayProperty=nameWithType> method:
```

Comment:
It is odd that we no longer document the old Startup class pattern in our app startup doc. We should probably fix that. But I think we can assume the reader is familiar with it if they care to read this section.

```suggestion
```

Reply:
I did have this here for visibility to those who have never seen it or forgotten about it.